recipe_scrapers/monsieurcuisine.py

```diff
@@ -1,72 +1,120 @@
 # mypy: disallow_untyped_defs=False
+import json
+import re
+
+import requests
+
 from ._abstract import AbstractScraper
-from ._utils import get_minutes, get_yields, normalize_string
+from ._utils import get_yields
+
+SCRIPT_PATTERN = re.compile(r'"recipeId":(\d+)')
 
 
 class MonsieurCuisine(AbstractScraper):
+    def __init__(self, url, proxies=None, timeout=None, *args, **kwargs):
+        super().__init__(url=url, proxies=proxies, timeout=timeout, *args, **kwargs)
+        scripts = self.soup.find_all("script")
+        recipe_id = None
+        for script in scripts:
+            matches = SCRIPT_PATTERN.search(str(script.string))
+            if matches:
+                recipe_id = matches.group(1)
+        language_iso = self.soup.find("html")["lang"]
+        data_url = f"https://mc-api.tecpal.com/api/v2/recipes/{recipe_id}"
+        self.page_data = requests.get(
+            data_url,
+            headers={"Accept-Language": language_iso, "Device-Type": "web"},
+            proxies=proxies,
+            timeout=timeout,
+        ).content
+        self.data = json.loads(self.page_data)
+
     @classmethod
     def host(cls):
         return "monsieur-cuisine.com"
 
     def author(self):
-        container = self.soup.find("p", {"class": "recipe_author"})
-
-        return container.span.get_text().strip()
+        return self.data.get("data").get("recipe").get("author").get("name")
 
     def title(self):
-        return self.soup.h1.get_text()
+        return self.data.get("data").get("recipe").get("name")
 
     def total_time(self):
-        container = self.soup.find("div", {"class": "duration-information"}).find(
-            "div", {"class": "recipe-duration-total"}
-        )
+        duration = self.data.get("data").get("recipe").get("duration")
```

Comment:
This can be converted into a `return self.data.get(...)` statement I think (we can skip the use of a `duration` variable).

Reply:
Yes good idea.